src/main/java/com/artipie/metrics/memory/InMemoryMetrics.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.metrics.memory;
+
+import com.artipie.metrics.Metrics;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * {@link Metrics} implementation storing data in memory.
+ *
+ * @since 0.9
+ * @todo #231:30min Support gauges in InMemoryMetrics.
+ *  `InMemoryMetrics.gauge()` method implementation should get or create an `InMemoryGauge` by name
+ *  and store it. `InMemoryMetrics.counters()` method should be added
+ *  to create snapshot of existing gauges. Implementations are expected to be similar to counters.
+ */
+final class InMemoryMetrics implements Metrics {
+
+    /**
+     * Counters by name.
+     */
+    private final ConcurrentMap<String, InMemoryCounter> cnts = new ConcurrentHashMap<>();
+
+    @Override
+    public InMemoryCounter counter(final String name) {
+        return this.cnts.computeIfAbsent(name, ignored -> new InMemoryCounter());
+    }
+
+    @Override
+    public InMemoryGauge gauge(final String name) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Get counters snapshot.
+     *
+     * @return Counters snapshot.
+     */
+    public Map<String, InMemoryCounter> counters() {
```

Comment:
@olegmoz I think that this method should be added to the interface: we can always create another abstractions if later we realize that `Map<String, Counter>` is not suitable for all `Metrics` implementations. Same stands for `value()` methods in `Counter` and `Gauge`. 
Also, not inherited methods in implementations do not seem to be a good practice: it always leads to casting/operating specific implementations instead of the interfaces.

Reply:
@olenagerasimova there was similar concern in PR with interfaces, see #238 . `Metrics` interface is for reporting metrics and what we are doing with counters and gauges may differ a lot. For the start we are storing them in memory and will print periodically to the log. It will not be the case with other metric aggregators. The fact that we can read counters map is specific for `InMemoryMetrics` and does not seem to be needed in other implementations I an imagine now.

Reply:
@olegmoz ok, thanks for the explanation